src/main/java/org/jenkinsci/plugins/workflow/cps/FlowHead.java

```diff
@@ -117,6 +117,19 @@ void newStartNode(FlowStartNode n) throws IOException {
             this.head = execution.startNodes.push(n);
         }
         execution.storage.storeNode(head, false);
+
+        CpsThreadGroup c = CpsThreadGroup.current();
+        if (c !=null) {
```

Comment:
I'm leaving this in, in case we *do* get a  proper CpsThreadGroup down the road, so it does The Right Thing (tm).